Automattic/jetpack
# Premium Blocks: Handle core/cover block

This PR implements the upgrade banner for `core/cover` blocks. For this purpose, it adds a context that allows to pass data through the components. 

Fixes #

#### Changes proposed in this Pull Request:

* Add Upgrade feature to core/cover block

#### Does this pull request change what data or activity we track or use?


no

#### Testing instructions:

* Apply both patches in your sandbox: D47508-code and D46955-code
* Compile these files and update your sandbox with these changes
* Test in a Simple site with a Free Plan
* Sandbox the testing site.
* Add a `core/cover` block to the editor
* Confirm that the block doesn't show the banner in the editor canvas
<img src="https://user-images.githubusercontent.com/77539/89186416-24496800-d572-11ea-87b7-8f3af356ac3f.png" width="500px" />

* Try to upload a video file either dropping on the media placeholder area or uploading by clicking on the `Upload` button. Confirm it shows the upgrade banner.

<img src="https://user-images.githubusercontent.com/77539/89215712-853b6500-d59f-11ea-82ea-37383fd63c7d.png" width="500px" />


* Try to set a background. It should work as expected.
<img src="https://user-images.githubusercontent.com/77539/89186699-8a35ef80-d572-11ea-89e6-a2f6dcf0c81c.png" width="500px" />

* Try to replace the background using the `Replace` control button from the block toolbar. It should act in the similar way that before.
<img src="https://user-images.githubusercontent.com/77539/89186778-a6399100-d572-11ea-85d0-c7a70c5bedb2.png" width="300px" />
